version.gradle.kts

```diff
@@ -24,19 +24,14 @@
  * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
-final def versions = [
-        errorProne      : "2.4.0",
-        errorProneJavac : "9+181-r4173-1", // taken from here: https://github.com/tbroyer/gradle-errorprone-plugin/blob/v0.8/build.gradle.kts
-        guava           : "31.0.1-jre",
-        checkerFramework: "3.21.0",
-        grpc            : "1.28.1",
-        pmd             : "6.41.0",
+val guavaVersion: String by extra("31.0.1-jre")
```

Comment:
Can we make the declaration of versions more conceptual? How about declaring them per-library and in `buildSrc` similar to how we do in the Spine libraries?

I don't like the idea of declaring the versions like this, as it looks hardly structured at the moment. And we will definitely add much more dependencies in the nearest future.